File: node/generate-pdf/src/pdf.js
import { PDFDocument } from 'pdf-lib';
import { Buffer } from 'node:buffer';

export async function createPdf({ id, date, name, items, total }) {
  const document = await PDFDocument.create();
  const page = document.addPage([595.28, 841.89]); // A4 size

  page.drawText('Sample Invoice', { x: 50, y: 750, size: 20 });
  page.drawText(new Date(date).toLocaleDateString(), {
    x: 400,
    y: 750,
    size: 15,
  });

  page.drawText(`Hello, ${name}!`, {
    x: 50,
    y: 700,
    size: 30,
  });

  page.drawText(`Order ID: ${id}`, {
    x: 50,
    y: 650,
    size: 10,
  });

  page.drawText(`Total: $${total}`, { x: 50, y: 600, size: 15 });

  const orderList = items
    .map(
      ({ description, quantity, cost }) =>
        `${description} x ${quantity} = $${cost}`
    )
    .join('\n');

  page.drawText(orderList, { x: 50, y: 550, size: 15 });

  const pdfBytes = await document.save();

  return Buffer.from(pdfBytes);
}

File: python/generate-pdf/src/pdf.py
import pdfkit

def create_pdf(order):
    content = """
    <h2>Sample Invoice</h2>
    <p>Date: {}</p>
    <h3>Hello, {}!</h3>
    <p>Order ID: {}</p>
    <p>Total: ${:.2f}</p>
    <ul>
    """.format(order["date"], order["name"], order["id"], order["total"])

    for item in order["items"]:
        content += "<li>{} x {} = ${:.2f}</li>".format(item["description"], item["quantity"], item["cost"])

    content += "</ul>"

    pdf_bytes = pdfkit.from_string(content, False)
    return pdf_bytes
